libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -456,8 +457,24 @@ void DeclarationTypeChecker::endVisit(VariableDeclaration const& _variable)
 	}
 	else if (_variable.isStateVariable())
 	{
-		solAssert(varLoc == Location::Unspecified, "");
-		typeLoc = (_variable.isConstant() || _variable.immutable()) ? DataLocation::Memory : DataLocation::Storage;
+		switch (varLoc)
+		{
+			case Location::Unspecified:
+				typeLoc = (_variable.isConstant() || _variable.immutable()) ? DataLocation::Memory : DataLocation::Storage;
+				break;
+			case Location::Transient:
+				if (_variable.isConstant() || _variable.immutable())
+					m_errorReporter.declarationError(
+						2197_error,
+						_variable.location(),
+						"Transient cannot be used as data location for constant or immutable variables."
+					);
+				typeLoc = DataLocation::Transient;
+				break;
+			default:
+				solAssert(false, "");
+				break;
+		}
```

Comment:
```suggestion
		switch (varLoc)
		{
		case Location::Unspecified:
			typeLoc = (_variable.isConstant() || _variable.immutable()) ? DataLocation::Memory : DataLocation::Storage;
			break;
		case Location::Transient:
			if (_variable.isConstant() || _variable.immutable())
				m_errorReporter.declarationError(
					2197_error,
					_variable.location(),
					"Transient cannot be used as data location for constant or immutable variables."
				);
			typeLoc = DataLocation::Transient;
			break;
		default:
			solAssert(false, "");
			break;
		}
```

I can't believe @cameel's not noticed this; although looking at the rest of the `DeclarationTypeChecker`, it's a mish mash of styles (even though the one I suggested is our "official" one), so consider this one highly optional.

Reply:
Yeah, that's right. Although it is indeed our style, every other `switch ... case` block in `DeclarationTypeChecker.cpp` is indented like that...
I am inclined to just leave it as is then since you recommended it as optional.

Reply:
Yeah, I'd tend to agree, as I prefer the style you already have than our official one :)